internal/services/mssql/mssql_server_resource.go

```diff
@@ -486,203 +488,166 @@ func resourceMsSqlServerRead(d *pluginsdk.ResourceData, meta interface{}) error
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ServerID(d.Id())
+	id, err := commonids.ParseSqlServerID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, "")
+	resp, err := client.Get(ctx, *id, servers.DefaultGetOperationOptions())
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
-			log.Printf("[INFO] Error reading SQL Server %s - removing from state", id.String())
+		if response.WasNotFound(resp.HttpResponse) {
+			log.Printf("[INFO] Error retrieving SQL Server %s - removing from state", id)
 			d.SetId("")
 			return nil
 		}
 
-		return fmt.Errorf("reading SQL Server %s: %v", id.Name, err)
+		return fmt.Errorf("retrieving SQL Server %s: %v", id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	if location := resp.Location; location != nil {
-		d.Set("location", azure.NormalizeLocation(*location))
-	}
-	identity, err := flattenSqlServerIdentity(resp.Identity)
-	if err != nil {
-		return fmt.Errorf("setting `identity`: %+v", err)
-	}
+	d.Set("name", id.ServerName)
+	d.Set("resource_group_name", id.ResourceGroupName)
 
-	if err := d.Set("identity", identity); err != nil {
-		return fmt.Errorf("setting `identity`: %+v", err)
-	}
+	if model := resp.Model; model != nil {
+		// NOTE: In the new API this just a string instead of a *string, shouldn't we just call Normalize?
+		d.Set("location", location.NormalizeNilable(pointer.To(model.Location)))
```

Comment:
indeed, this can become:

```suggestion
		d.Set("location", location.Normalize(model.Location))
```

Reply:
Fixed.